.github/workflows/lint-scripts/websites-shared-credentials-sort-order.rb

```diff
@@ -1,14 +1,27 @@
 require 'json'
 
-shared_websites = JSON.parse(File.read('quirks/websites-with-shared-credential-backends.json'))
+def process_file(file_path)
+  shared_websites = JSON.parse File.read(file_path)
+  shared_websites_sorted = shared_websites.sort do |a, b|
+    a_string = a["shared"] ? a["shared"].first : (a["from"] ? a["from"].first : "")
+    b_string = b["shared"] ? b["shared"].first : (b["from"] ? b["from"].first : "")
+    a_string <=> b_string
+  end
 
-shared_websites_sorted = shared_websites.sort_by(&:first)
+  File.open(file_path, 'w') do |f|
+    f.write JSON.pretty_generate(shared_websites_sorted, indent: '    ')
+  end
 
-File.open('quirks/websites-with-shared-credential-backends.json','w') do |f|
-  f.write(JSON.pretty_generate(shared_websites_sorted, indent: '    '))
+  unless shared_websites == shared_websites_sorted
+    STDERR.puts "The JSON in '#{file_path}' isn't sorted!"
+    return false
```

Comment:
I'm sorta wondering what the point of logging to `stderr` and returning `false` is here if this script will read, sort, and write the contents of the file regardless of those signals. What happens elsewhere if this script returns `false`? Does a build or commit step fail?

Reply:
We use the `false` return value to signal to the overall script that we should “fail” the script.

```
shared_credentials_were_sorted = process_file "quirks/shared-credentials.json"
shared_credentials_historical_were_sorted = process_file "quirks/shared-credentials-historical.json"
if !shared_credentials_were_sorted or !shared_credentials_historical_were_sorted
  exit 1
end
```

This is useful for our Github actions that’ll tell a committer on their own fork if they’ve sorted things right. So we’re doing two things:

1. Failing in the CI so that the contributor knows something went wrong.
2. Writing out the fixed version, so that if it’s run locally, it’s easy to fix. :)

Reply:
Ah, that second part is helpful context. 🙂 Thanks!